django/contrib/contenttypes/management.py

```diff
@@ -1,9 +1,56 @@
 from django.apps import apps
-from django.db import DEFAULT_DB_ALIAS, router
+from django.db import DEFAULT_DB_ALIAS, migrations, router, transaction
+from django.db.utils import IntegrityError
 from django.utils import six
 from django.utils.six.moves import input
 
 
+class RenameContentType(migrations.RunPython):
+    def __init__(self, app_label, old_name, new_name):
+        self.app_label = app_label
+        self.old_name = old_name
+        self.new_name = new_name
+        super(RenameContentType, self).__init__(self.rename_forward, self.rename_backward)
+
+    def _rename(self, apps, old_name, new_name):
+        ContentType = apps.get_model('contenttypes', 'ContentType')
+        try:
+            content_type = ContentType.objects.get_by_natural_key(self.app_label, old_name.lower())
+        except ContentType.DoesNotExist:
+            pass
+        else:
+            content_type.model = new_name.lower()
+            using = router.db_for_write(ContentType, instance=content_type)
+            try:
+                with transaction.atomic(using=using):
+                    content_type.save(update_fields={'model'})
+            except IntegrityError:
```

Comment:
I'm not convinced failing hard by reraising the exception would be appropriate here.

Think of all the users who installed `'django.contrib.contenttypes'` only to get the admin working and ended up with a lot of stale contenttypes because they ran `migrate` with `--noinput` during an automated deployment or simply never deleted their stale content type because they had no idea what they were prompted for.

I suppose these users would be really confused if such an exception would be raised after they simply tried to rename a model. It gets even worse if you take into account databases that can't rollback DDL statement; users would be on they own trying to figure out what happened with a recorded state different from the actual database state.

IMO warning the user with a `RuntimeWarning` or `print() if verbosity >= 1` would be more appropriate.